File: BuilderDesignPattern/CarManufacture_Example/Components/Engine.java
package BuilderDesignPattern.CarManufacture_Example.Components;

public class Engine {
	private final int volume;
	private long mileage;
	private boolean started;

	public Engine(int engineVolume, long engineMileage) {
		this.volume = engineVolume;
		this.mileage = engineMileage;
	}

	public void turnOn() {
		this.started = true;
	}

	public void turnOff() {
		this.started = false;
	}

	public boolean isStarted() {
		return started;
	}

	public void go(long miles) {
		if (this.started) {
			this.mileage += miles;
		} else {
			System.out.println("Cannot go(). First Start the Car.");
		}
	}

	public int getVolume() {
		return this.volume;
	}

	public long getMileage() {
		return this.mileage;
	}
}

File: BuilderDesignPattern/CarManufacture_Example_Python/Components/Engine.py
from typing import Final

class Engine:

	_volume: Final[int]
	_mileage: int
	_started: bool

	def __init__(self, engineVolume, engineMileage):
		self._volume = engineVolume
		self._mileage = engineMileage
		self._started = False

	def _isStarted(self):
		if self._started:
			return True
		else:
			return False

	def turnOn(self):
		if self._isStarted():
			print(f"The Car is already on.")
		else:
			self._started = True
			print(f"The Car is Turned On")

	def turnOff(self):
		if self._isStarted():
			self._started = False
			print(f"The Car is turned Off")
		else:
			print(f"The Car is already off")

	def go(self, miles: int):
		if self._isStarted():
			self._mileage += miles
		else:
			print(f"Start the car First")

	def getVolume(self):
		return self._volume

	def getMileage(self):
		return self._mileage